src/data/finset/sups.lean

```diff
@@ -0,0 +1,288 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.finset.n_ary
+
+/-!
+# Set family operations
+
+This file defines a few binary operations on `finset α` for use in set family combinatorics.
+
+## Main declarations
+
+* `finset.sups s t`: Finset of elements of the form `a ⊔ b` where `a ∈ s`, `b ∈ t`.
+* `finset.infs s t`: Finset of elements of the form `a ⊓ b` where `a ∈ s`, `b ∈ t`.
+* `finset.disj_sups s t`: Finset of elements of the form `a ⊔ b` where `a ∈ s`, `b ∈ t` and `a`
+  and `b` are disjoint.
+
+## Notation
+
+We define the following notation in locale `finset_family`:
+* `s ⊻ t` for `finset.sups s t`
+* `s ⊼ t` for `finset.infs s t`
+* `s ○ t` for `finset.disj_sups s t`
```

Comment:
Is this notation ubiquitous? The univode characters have names that don't match your reference, so my feeling would be you should probably add something to `references.bib` to back up the choices.

Reply:
That's the notation Bollobás used in his Combinatorics course this year.